compiler/rustc_mir_transform/src/uninhabited_enum_branching.rs

```diff
@@ -71,6 +72,28 @@ fn variant_discriminants<'tcx>(
     }
 }
 
+/// Ensures that the `otherwise` branch leads to an unreachable bb, returning `None` if so and a new
+/// bb to use as the new target if not.
+fn ensure_otherwise_unreachable<'tcx>(
+    body: &Body<'tcx>,
+    targets: &SwitchTargets,
+) -> Option<BasicBlockData<'tcx>> {
+    let otherwise = targets.otherwise();
+    let bb = &body.basic_blocks()[otherwise];
+    if bb.statements.len() == 0
+        && bb.terminator().kind == TerminatorKind::Unreachable
```

Comment:
I always wonder about StorageLive/Dead in these.

Would it make sense for this to be, say,
```rust
    if bb.terminator().kind == TerminatorKind::Unreachable
        && !bb.is_cleanup
        && bb.statements.iter().all(|s| matches!(&s.kind, StatementKind::StorageDead(_))) {
```

Reply:
Probably? I'll wait for Wesley to take a look

Reply:
I've changed this for now since I was fixing the other thing, but can revert this if there's a reason to prefer the other way around